test/bench/bench.cpp

```diff
@@ -0,0 +1,243 @@
+        benchmark::DoNotOptimize(result);
+    }
+}
+
+void load_benchmark(const fs::path& path, const std::string& name_prefix)
+{
+    const auto base_name = name_prefix + path.stem().string();
+
+    std::ifstream wasm_file{path};
+    const auto wasm_binary = std::make_shared<const fizzy::bytes>(
+        std::istreambuf_iterator<char>{wasm_file}, std::istreambuf_iterator<char>{});
+
+    benchmark::RegisterBenchmark(("parse/" + base_name).c_str(),
+        [wasm_binary](benchmark::State& state) { benchmark_parse(state, *wasm_binary); });
+
+    benchmark::RegisterBenchmark(("instantiate/" + base_name).c_str(),
+        [wasm_binary](benchmark::State& state) { benchmark_instantiate(state, *wasm_binary); });
+
+    enum class InputsReadingState
+    {
+        Name,
+        FuncName,
+        FuncArguments,
+        Memory,
+        ExpectedResult,
+        ExpectedMemory,
+    };
+
+    const auto inputs_path = fs::path{path}.replace_extension(InputsExtension);
+    if (fs::exists(inputs_path))
+    {
+        auto st = InputsReadingState::Name;
+        auto inputs_file = std::ifstream{inputs_path};
+        std::string input_name;
+        ExecutionBenchmarkCase benchmark_case;
+        for (std::string l; std::getline(inputs_file, l);)
+        {
+            switch (st)
+            {
+            case InputsReadingState::Name:
+                if (l.empty())
+                    continue;
+                input_name = std::move(l);
+                benchmark_case = {};
+                benchmark_case.wasm_binary = wasm_binary;
+                st = InputsReadingState::FuncName;
+                break;
+
+            case InputsReadingState::FuncName:
+                benchmark_case.func_name = std::move(l);
+                st = InputsReadingState::FuncArguments;
+                break;
+
+            case InputsReadingState::FuncArguments:
+            {
+                std::istringstream iss{l};
+                std::copy(std::istream_iterator<uint64_t>{iss}, std::istream_iterator<uint64_t>{},
+                    std::back_inserter(benchmark_case.func_args));
+                st = InputsReadingState::Memory;
+                break;
+            }
+
+            case InputsReadingState::Memory:
+                benchmark_case.memory = fizzy::from_hex(strip_space(l));
+                st = InputsReadingState::ExpectedResult;
+                break;
+
+            case InputsReadingState::ExpectedResult:
+                if (!l.empty())
+                    benchmark_case.expected_result = std::stoull(l);
+                st = InputsReadingState::ExpectedMemory;
+                break;
+
+            case InputsReadingState::ExpectedMemory:
+                benchmark_case.expected_memory = fizzy::from_hex(strip_space(l));
+
+                benchmark::RegisterBenchmark(("execute/" + base_name + '/' + input_name).c_str(),
+                    [benchmark_case](
```

Comment:
I think you could move into lambda like
```
[case = std::move(benchmark_case)](benchmark::State& state) { benchmark_execute(state, case); }
```
And then reset it right after `RegisterBenchmark`
```
benchmark_case = {};
```